clients/ui/frontend/src/app/pages/modelCatalogSettings/components/ManageSourceForm.tsx

```diff
@@ -0,0 +1,134 @@
+import { FORM_LABELS, DESCRIPTIONS } from '~/app/pages/modelCatalogSettings/constants';
+import SourceDetailsSection from './SourceDetailsSection';
+import CredentialsSection from './CredentialsSection';
+import YamlSection from './YamlSection';
+import ModelVisibilitySection from './ModelVisibilitySection';
+import PreviewPanel from './PreviewPanel';
+import ManageSourceFormFooter from './ManageSourceFormFooter';
+
+type ManageSourceFormProps = {
+  existingData?: Partial<ManageSourceFormData>;
+  isEditMode: boolean;
+};
+
+const ManageSourceForm: React.FC<ManageSourceFormProps> = ({ existingData, isEditMode }) => {
+  const navigate = useNavigate();
+  const [formData, setData] = useManageSourceData(existingData);
+  const [isFiltersExpanded, setIsFiltersExpanded] = React.useState(false);
+  const [isSubmitting, setIsSubmitting] = React.useState(false);
+  const [submitError, setSubmitError] = React.useState<Error | undefined>(undefined);
+
+  const isHuggingFaceMode = formData.sourceType === SourceType.HuggingFace;
+  const isFormComplete = isFormValid(formData);
+  const canPreview = isPreviewReady(formData);
+
+  const handleSubmit = async () => {
+    setIsSubmitting(true);
+    setSubmitError(undefined);
+
+    try {
+      // TODO: Implement submit logic (will be part of API integration)
+      // navigate(catalogSettingsUrl());
+    } catch (error) {
+      setSubmitError(error instanceof Error ? error : new Error('Failed to save source'));
+    } finally {
+      setIsSubmitting(false);
+    }
+  };
+
+  const handlePreview = () => {
+    // TODO: Implement preview logic (will be part of API integration)
+  };
+
+  const handleCancel = () => {
+    navigate(catalogSettingsUrl());
+  };
+
+  return (
+    <>
+      <Sidebar hasGutter hasBorder isPanelRight>
+        <SidebarContent>
+          <Form isWidthLimited>
+            <Stack hasGutter>
+              <StackItem>
+                <SourceDetailsSection formData={formData} setData={setData} />
+              </StackItem>
+
+              {isHuggingFaceMode && (
+                <StackItem>
+                  <CredentialsSection formData={formData} setData={setData} />
+                </StackItem>
+              )}
+
+              {!isHuggingFaceMode && (
+                <StackItem>
+                  <YamlSection formData={formData} setData={setData} />
+                </StackItem>
+              )}
+
+              <StackItem>
+                <ModelVisibilitySection
+                  formData={formData}
+                  isExpanded={isFiltersExpanded}
+                  onToggle={() => setIsFiltersExpanded((prev) => !prev)}
+                  setData={setData}
+                />
+              </StackItem>
+
+              <StackItem>
+                <FormSection>
+                  <FormGroup fieldId="enable-source">
+                    <Checkbox
+                      label={<strong>{FORM_LABELS.ENABLE_SOURCE}</strong>}
```

Comment:
`<strong>` is actually too bold 😆 it isn't looking consistent with the other labels - sorry, can you actually replace it with a `<span className="pf-v6-c-form__label-text">`? that should make it look consistent with the other form labels (I guess it's a medium-weight font)